src/libraries/System.Data.OleDb/src/OleDbCommand.cs

```diff
@@ -411,14 +411,40 @@ private void CreateAccessor()
             _dbBindings = bindings;
         }
 
-        private void ApplyParameterBindings(UnsafeNativeMethods.ICommandWithParameters commandWithParameters, tagDBPARAMBINDINFO[] bindInfo)
+        private unsafe void ApplyParameterBindings(UnsafeNativeMethods.ICommandWithParameters commandWithParameters, tagDBPARAMBINDINFO[] bindInfo)
         {
             IntPtr[] ordinals = new IntPtr[bindInfo.Length];
             for (int i = 0; i < ordinals.Length; ++i)
             {
                 ordinals[i] = (IntPtr)(i + 1);
             }
-            OleDbHResult hr = commandWithParameters.SetParameterInfo((IntPtr)bindInfo.Length, ordinals, bindInfo);
+
+            IntPtr bindingPtr;
+
+            if (ODB.IsRunningOnX86)
+            {
+                tagDBPARAMBINDINFO_x86[] bindInfo_x86 = new tagDBPARAMBINDINFO_x86[bindInfo.Length];
+                for (int i = 0; i < bindInfo.Length; i++)
+                {
+                    fixed (tagDBPARAMBINDINFO* p = &bindInfo[i])
+                    {
+                        bindInfo_x86[i] = *(tagDBPARAMBINDINFO_x86*)p;
+                    }
+                }
+                fixed (tagDBPARAMBINDINFO_x86* p = &bindInfo_x86[0])
+                {
+                    bindingPtr = (IntPtr)p;
+                }
+            }
+            else
+            {
+                fixed (tagDBPARAMBINDINFO* p = &bindInfo[0])
+                {
+                    bindingPtr = (IntPtr)p;
```

Comment:
The pinned pointer is valid only inside the fixed block. It is not ok to use it outside the fixed block.

The last commit should be reverted.

Reply:
Done